### DISCUSSION TITLE
Is there any documentation on the use of governorFacet?

### DISCUSSION THREAD

**Discussion by kbennett2000** [Agoric/agoric-sdk#8834] [General]
Currently we don't have any documentation on governorFacet. Users in the community have asked about it, it would be great to have some resources to point them to!

**Comment by dillu24** [comment]
## Some context

Let's say I have a contract with some parameters that I want configured from time-to-time. For proper decentralization ideally this should be configurable by governance instead of the contract creator.

Ex. a contract which communicates through IBC with another chain might need IBC timeout parameters

**Reply by 0xpatrickdev** [reply]
The `@agoric/governance` package provides a general framework for this. The [README](https://github.com/Agoric/agoric-sdk/blob/master/packages/governance/README.md) has a detailed summary, and there are some [helpful diagrams](https://github.com/Agoric/agoric-sdk/blob/master/packages/governance/docs/index.md) as well.

The idea is to extract capabilities that might be in a `creatorFacet` and allow them to be controlled by a committee.  Some of these capabilities might be: 
 - changes to declared parameters
 - invocation of declared methods
 - blocking exercise of some invitations
 
The contract can leverage different counters, like `BinaryVoteCounter`, to tally votes and ensure committee quorum is met.

The Inter protocol contracts leverage this governance framework - the deployed instances are `economicCommittee` and `econCommitteeCharter`. `KREAd`, the first third-party dapp also makes use of this contract: [kreadCommitteeCharter.js](https://github.com/Kryha/KREAd/blob/54b3438334dbd9d1130764badac0a7a3c2de938f/agoric/contract/src/kreadCommitteeCharter.js), [kread-committee-proposal.js](https://github.com/Kryha/KREAd/blob/develop/agoric/contract/src/proposal/kread-committee-proposal.js).

**Reply by migueldingli1997** [reply]
Thanks for the reply @0xpatrickdev . Just to make sure I understand - when we refer to governance in this context, we are not referring to standard (BLD) governance, right?

**Reply by 0xpatrickdev** [reply]
> when we refer to governance in this context, we are not referring to standard (BLD) governance, right?

Correct, this is separate from that.

The committee would be invited during the CoreEval proposal - see the `voterAddresses` param in the [kread-committee-proposal](https://github.com/Kryha/KREAd/blob/develop/agoric/contract/src/proposal/kread-committee-proposal.js). BLD governance needs to approve the initial CoreEval, but once it passes, `voterAddresses` are in control of the delegated capabilities.

**Reply by 0xpatrickdev** [reply]
Two more things worth noting - 

1. It's expected that consumers of `@agoric/governance` leverage the _installation_ already present on chain and start their own _instance_ of `committee`. The implementation is intended to support a multitude of use cases, and includes important functionality like `replaceElectorate` that may be forgotten in userland implementations. _If the contract doesn't support a use case you have, it'd be great to report back here!_

2. There's a CLI available via `agops gov`:
```bash
$ alias gov="yarn run --silent agops gov"
$ gov --help

Usage: agops gov|ec [options] [command]

Electoral governance commands

Options:
  --keyring-backend <os|file|test>  keyring's backend (os|file|test) (default "test") (default: "test")
  -h, --help                        display help for command

Commands:
  committee [options]               accept invitation to join a committee
  charter [options]                 accept the charter invitation
  find-continuing-id [options]      print id of specified voting continuing invitation
  find-continuing-ids [options]     print records of voting continuing invitations
  vote [options]                    vote on latest question
  proposePauseOffers [options]      propose a vote to pause offers
  help [command]                    display help for command
 ```
